### DISCUSSION TITLE
[GUIDE NEEDED] How to get wallet provider from Appkit v5.1.5

### DISCUSSION THREAD

**Discussion by ms-safiq** [reown-com/appkit#2793] [General]
Hi,

I recently migrated web3modal/ethers from v5.0.11 to v5.1.5. Seems like there were some method removed in the latest version. I was using modal.subscribeProvider(...) or modal.getWalletProvider() and these are removed from the latest version.

I am unable the find the details in the doc as well. The appkit documentation is not updated.

Please help me with how to get the walletprovider from the connected wallet in the appkit v5.1.5?

I was using 5.0.11 and it was throwing Third-party cookie will be blocked in future Chrome versions as part of Privacy Sandbox. (10000+ times). So i migrated to appkit 5.1.5 and ended up losing the access to subscribeProvider.

@enesozturk

**Comment by rtomas** [comment]
please update to v.5.1.6. 
https://github.com/WalletConnect/web3modal/discussions/2763#discussioncomment-10540175

**Reply by ms-safiq** [reply]
@rtomas How do we disconnect? There is not diconnect method like before.

**Reply by ms-safiq** [reply]
@rtomas How do we disconnect the socials? I am using below code to disconnect the wallet connect, but however the social login is trying to reconnect automatically and providing the address of the social login and treating it as logged in user though i execute the below method. This issue is not happening with the metamask or any other wallets. 

```
disconnectWalletConnect() {
    if (this.wcModal) {
      if (this.wcModal.adapters) {
        this.wcModal.adapters[0].connectionControllerClient?.disconnect();
      }
      if (this.wcModal.adapter) {
        this.wcModal.adapter.connectionControllerClient?.disconnect();
      }
    }
  }
```

**Reply by rtomas** [reply]
I have just tried in version 1.0.7 and i cannot reproduce this. can you confirm to me ?
do you have a minimal example ? are you using wagmi, ethers, solana ?
